streams/src/main/java/com/palantir/common/streams/KeyedStream.java

```diff
@@ -46,6 +46,12 @@
  */
 public interface KeyedStream<K, V> {
 
+    /**
+     * Returns a keyed stream consisting of elements in this stream,
+     * performing the provided action on each element of this stream.
+     */
+    KeyedStream<K, V> peek(BiConsumer<K, V> consumer);
```

Comment:
Some risk if there are external implementations of `KeyedStream`, since this is a non-default method. Perhaps this should do something along the lines of `return KeyedStream.ofEntries(entries().peek(entry ->consumer.accept(entry.getKey(), entry.getValue()));`? What do you think?

Reply:
Can do!

Reply:
I can't do `KeyedStream.ofEntries` because `entries()` exposes `Map.Entry<? extends K, ? extends V>`. Using impl in interface seems weird, but seems to be what we do somewhere else in this class anyway? See `KeyedStream#join` for example.